packages/flutter/test/widgets/editable_text_cursor_test.dart

```diff
@@ -1016,128 +1016,90 @@ void main() {
-              keyboardType: TextInputType.text,
-              onChanged: (String value) {
-                changedValue = value;
-              },
-              cursorWidth: 15.0,
-              cursorHeight: 30.0,
-            ),
-          ],
+    expect(findRenderEditable(tester), paints..rect(
+      rect: const Rect.fromLTWH(0.0, 0.0, 2.0, expectedCursorHeight),
+    ));
+  });
+
+  testWidgets('Material3 - Default cursor layout has correct height and vertical offset on iOS and macOS', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Material(
+          child: TextField(autofocus: true),
         ),
       ),
     );
-    await tester.pumpWidget(widget);
 
-    // Populate a fake clipboard.
-    const String clipboardContent = 'Hello world!';
-    tester.binding.defaultBinaryMessenger.setMockMethodCallHandler(SystemChannels.platform, (MethodCall methodCall) async {
-      if (methodCall.method == 'Clipboard.getData') {
-        return const <String, dynamic>{'text': clipboardContent};
-      }
-      if (methodCall.method == 'Clipboard.hasStrings') {
-        return <String, dynamic>{'value': clipboardContent.isNotEmpty};
-      }
-      return null;
-    });
+    // Default style has a font size of 16 and a line height of 1.5, so preferredLineHeight is 24.
+    // On macOS and iOS the cursor is extended by 2.0 for fidelity.
+    const double cursorReduction = 2.0;
+    const double expectedCursorHeight = 24.0 - cursorReduction;
+    // RenderEditable.getLocalRectForCaret applies a vertical offset on Apple platforms.
+    const double verticalOffset = 2.0;
 
-    // Long-press to bring up the text editing controls.
-    final Finder textFinder = find.byKey(editableTextKey);
-    await tester.longPress(textFinder);
-    tester.state<EditableTextState>(textFinder).showToolbar();
-    await tester.pumpAndSettle();
+    expect(findRenderEditable(tester), paints..rrect(
+      rrect: RRect.fromRectAndRadius(
+        const Rect.fromLTWH(0.0, (verticalOffset + cursorReduction) / 2, 2.0, expectedCursorHeight),
+        const Radius.circular(2)
+      )
+    ));
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS,  TargetPlatform.macOS }));
 
-    await tester.tap(find.text('Paste'));
-    await tester.pump();
+  testWidgets('Material3 - Default cursor layout has correct height and vertical offset', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Material(
+          child: TextField(autofocus: true),
+        ),
+      ),
+    );
 
-    expect(changedValue, clipboardContent);
+    // Default style has a font size of 16 and a line height of 1.5, so preferredLineHeight is 24.
+    // On macOS and iOS the cursor is extended by 2.0 for fidelity.
+    const double expectedCursorHeight = 24.0;
 
-    await expectLater(
-      find.byKey(const ValueKey<int>(1)),
-      matchesGoldenFile('editable_text_test.3.png'),
```

Comment:
Are we really getting rid of these goldens?

Reply:
I think they should be replaced by proper check of what is painted, otherwise it is impossible to know what was the expected value.

Looking more closely, I correctly replaced the test which was generating `editable_text_test.2.png`.
But I should add a test to replace the one which was generating `editable_text_test.3.png` because this one was setting the cursor height to 30 explicitely. I will do it on Monday.

Reply:
I have updated the PR to add a test similar to the one I replaced.